### PR TITLE
feat(pageFilters): Store pinned page filter state

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Stores the pin state in local storage

### CODE REVIEW

**Review by davidenwang — COMMENTED** [review]
hmmm how will this work in a case for example we pin the projects filter to persist certain projects in local storage and then change the project filter selection. Does it stay pinned with subsequent changes overriding local storage, or does changing the selection unpin?

**Comment by evanpurkhiser** [comment]
So right now local storage essentially is the 'pinned state' for things.

That means anytime anything 'saves' to local storage, that will be used for the pinned state. So when switching projects, that updates the 'pinned' state.

Other things may _not_ updated pinned state though. 

I'll be making an PR here in a few that will restore the local storage state _only_ when it's pinned (if you have the feature flag).

**Review by davidenwang — APPROVED** [review]

File: static/app/actionCreators/pageFilters.tsx [hunk 1]
```diff
@@ -8,15 +8,14 @@ import * as qs from 'query-string';
 
 import PageFiltersActions from 'sentry/actions/pageFiltersActions';
 import {getStateFromQuery} from 'sentry/components/organizations/pageFilters/parse';
-import {PageFiltersStringified} from 'sentry/components/organizations/pageFilters/types';
 import {
-  getDefaultSelection,
   getPageFilterStorage,
   setPageFiltersStorage,
-} from 'sentry/components/organizations/pageFilters/utils';
+} from 'sentry/components/organizations/pageFilters/persistance';
```
Comment (davidenwang) [inline static/app/actionCreators/pageFilters.tsx:14]: nit: persistence*

**Review by davidenwang — APPROVED** [review]

File: static/app/actionCreators/pageFilters.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/actionCreators/pageFilters.tsx:14]: damnit lol

**Pull request merged**